http_parser.cc

```diff
@@ -0,0 +1,104 @@
+#include "http_parser.h"
+#include <string>
+#include <vector>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+HttpParser* HttpParser::MakeHttpParser(const char* const raw_req)
```

Comment:
i think this should be dealt with in connection.cc when boost reads the request into the data_ member variable